packages/@lwc/features/src/types.ts

```diff
@@ -86,6 +86,12 @@ export interface FeatureFlagMap {
      * properly.
      */
     LEGACY_LOCKER_ENABLED: FeatureFlagValue;
+
+    /**
+     * If true, then the legacy constructor check is only used if
+     * LEGACY_LOCKER_ENABLED is true.
+     */
+    DISABLE_LIGHTNING_CONSTRUCTOR_CHECK: FeatureFlagValue;
```

Comment:
```suggestion
    /**
     * A manual override for `LEGACY_LOCKER_ENABLED`; should not be used if that flag is correctly set. If true, behave as if legacy Locker is enabled. If false or unset, then the value of the `LEGACY_LOCKER_ENABLED` flag is used.
     */
    ENABLE_LEGACY_LOCKER_VALIDATION: FeatureFlagValue;
```
1. "Lightning constructor check" is an internal implementation detail. Nobody other than us will understand what it means, and it's scoped to a very particular thing. If we use a feature-level name then more people will understand it and we can re-use it if needed.
2. Disambiguate it from `LEGACY_LOCKER_ENABLED` because that's confusing.

Reply:
Good points. I'd say the flag should be `ENABLE_LEGACY_VALIDATION` though as the original validation was not locker related.